### DISCUSSION TITLE
what is CGAL_USE_BASIC_VIEWER ??

### DISCUSSION THREAD

**Discussion by PinkMushroom** [CGAL/cgal#8667] [Q&A] [answered]
I tried the example from the documentation

https://doc.cgal.org/latest/Manual/devman_create_and_use_a_cmakelist.html

I modified the CMakeLists.txt file to use Qt5 instead of Qt6.

Everything built successfully but when I run the program I get the following error

**Impossible to draw, CGAL_USE_BASIC_VIEWER is not defined.**

I cannot find anything in the documentation that discusses it.

**Comment by gdamiand** [comment]
Cf the example in the section  "Minimal Example Using Qt6" https://doc.cgal.org/latest/Manual/devman_create_and_use_a_cmakelist.html#secexample

CGAL_USE_BASIC_VIEWER is defined automatically if you link with the target CGAL::CGAL_Basic_viewer (and if this target exists).

**Reply by PinkMushroom** [reply]
I don't have much experience with cmake, I'm finding it quite hard to follow what exactly is going on to set up a build.

I'm just copying CMakeLists.txt file and hoping it works.

And it does work if I edit them correctly . LOL.

**Comment by PinkMushroom** [comment, marked as answer]
oh good grief. When I update the CMakeLists.txt file I forgot to update the "if(CGAL_Qt6_FOUND)" section.
Here's the complete file (the example is for Qt6, and my system uses Qt5)
```

cmake_minimum_required(VERSION 3.12...3.29)
project(Surface_mesh_Examples)
#CGAL_Qt5 is needed for the drawing.
find_package(CGAL REQUIRED OPTIONAL_COMPONENTS Qt5)
#create the executable of the application
create_single_source_cgal_program("draw_surface_mesh.cpp")
if(CGAL_Qt5_FOUND)
 
  #link it with the required CGAL libraries
  target_link_libraries(draw_surface_mesh PUBLIC CGAL::CGAL_Basic_viewer)
endif()
 
#end of the file
```
Example works now :-)
Sorry for the bother and thank you for the quick response !!

**Comment by For-Chance** [comment]
Maybe so?
```
find_package(CGAL CONFIG OPTIONAL_COMPONENTS Qt5)
target_link_libraries(PROJECTNAME PUBLIC CGAL::CGAL CGAL::CGAL_Qt5)
```
